Move text rendering to imGui

royfalk opened the pull request:
Text rendering will now use modern techniques with anti-aliasing and what not.

Please answer the following:

**Code Changes:**
- [ ] Have the PR Validation Tests been run? Not yet. I've focused exclusively on testing the test rendering, as that's what I've touched.

Issues:
- Mouse cursor is no longer the green arrow. May be unrelated to this PR.
- No support for font and size. Currently uses imGui default.
- No support for undocumented markup. See for example "<A Dead Man's Ship>" in the intro monologue.
- Load game menu - does not rely on the base computer code in the engine. Instead uses its own python code. Changes broke some of the functionality there. Font is default. Mouse over and selected not working.
- Text location is not perfect. It wasn't perfect before but now it's easier to see.
- Base computer (engine) still uses the old system. Not sure why.

stephengtuggy commented:
> On play testing, the following areas need to be fixed before considering a merge into master:
> 
>     1. Font size to follow what is set in the config.
> 
>     2. Base text highlighting should be proper for each hotspot.
> 
>     3. Able to load saved games by clicking on them.

To me, item 2 above is a lower priority, since it was already broken prior to this PR. Item 1 is a medium priority, a nice-to-have. Item 3 is a show-stopper though. We definitely need to be able to load saved games, including from the main menu.

royfalk commented:
> Okay, what's the path forward then?

I've removed the reliance on Roboto. For now, the code uses the default imGui font at size 18, which seems OK.
I've also added support for line width background. A bit of an intuition leap. The original width value is clearly a fraction of the screen. I used `float width = fraction * configuration().graphics.resolution_x * 2;` and it's bang on. <shrug>.
Still pending, implementation of font color and startlower. This is probably why some stuff in the cockpit is overlapping.

evertvorster reviewed (approved):
This code now runs, and I can load a saved game, so from a usability point of view this is good to go... with the understanding that we'll sort out font sizes changing with config options soon, and getting rid of that overlapping fonts in the VDU.

stephengtuggy commented on engine/src/gfxlib_struct.cpp:
```diff
@@ -36,13 +36,34 @@
 #include "configuration/configuration.h"
 
+
+// Implicit conversion FROM ImU32
```
Note: this isn't just me saying this. It's CLion and, I believe, clangd as well.

stephengtuggy replied:
Fixed. Thanks!

stephengtuggy commented on libraries/imgui/CMakeLists.txt:
```diff
@@ -28,6 +28,14 @@
 MESSAGE(STATUS "PROCESSING DIRECTORY: ${CMAKE_CURRENT_SOURCE_DIR}")
 
+#Find SDL_image
+FIND_PACKAGE(SDL2_IMAGE REQUIRED)
```
The root CMakeLists.txt doesn't already pull this in?

royfalk commented:
Some additional comments:
1. The whole text code is arcane. It looks like the position is bottom left. So in imgui I need to go up one line to be where I need to be.
2. Some layout in VDU is done via new lines. Something like:
   Line 1\nLine2\n\n\nLine 5
3.In load game, I can't make the selected game background to be the line width instead of text width. If I do, it breaks other things. Not going to bother with it. I'll just add this to the tasks and fix it right.
